hack/vmimage-validate.sh

```diff
@@ -3,9 +3,7 @@
 cleanup() {
   set +e
 
-  generate_artifacts
-
-  delete
+  az group delete -g "${BUILD_RESOURCE_GROUP}" --yes --no-wait
```

Comment:
Does `delete` do something more than this? Trying to understand the need for this change.

Reply:
@ehashman @charlesakalugwu 
Delete calls the delete function [here](https://github.com/openshift/openshift-azure/blob/master/hack/tests/ci-prepare.sh#L67).  This calls `make delete` which executes https://github.com/openshift/openshift-azure/blob/master/hack/delete.sh.

This fires up the fakerp and attempts to call delete cluster.  At this point in time we do not have a cluster only a vm running our image.  The fakerp then fails to execute because it doesn't find the files in `_data` directory.  Failing then triggers the `cleanup` trap on exit.  This fails the validate script.  Hopefully this helps clarify the process.

Reply:
thanks for this :)

Reply:
@kwoodson do we need to make similar changes to `hack/vmimage.sh`? I'm getting the following somewhere after `./hack/delete.sh mradchuk-tests`

> Resource group 'mradchuk-tests' could not be found.

Reply:
@m1kola 
I have seen this message.  I think this must happen in the `make delete` and maybe somewhere here:
https://github.com/openshift/openshift-azure/blob/master/pkg/util/azureclient/openshiftmanagedcluster/2019-04-30/client.go#L208

I believe this to be a symptom of calling delete and then waiting for the delete to finish and then returning the message that it is actually gone. 
```
$ az group delete -g ktest-noexist
Are you sure you want to perform this operation? (y/n): y
Resource group 'ktest-noexist' could not be found.
```

I think hitting this in the trap `cleanup` just ensures that its missing.  I'm not sure exactly where that message comes from.